codegen/cmd/injection-gen/generators/reconciler_reconciler.go

```diff
@@ -236,6 +236,10 @@ func (r *reconcilerImpl) Reconcile(ctx context.Context, key string) error {
 		}
 	}
 
+        // Since the Reconciler observed this object, update the Status.ObservedGeneration
+        // to match the objects ObservedGeneration.
+        resource.Status.ObservedGeneration = original.Generation
```

Comment:
I don't think that this alone is enough.  If `ReconcileKind` returned an error before updating `Status` then we might inadvertently convey readiness before we mean to by auto-bumping ObsGen before the conditions are (successfully) updated.  I would be supportive of doing this whenever the return code is not an error 🤷‍♂

Reply:
@n3wscott has a good point too, we [will] use this for a couple resources in Serving that don't touch status, and shouldn't update this.  I'd be ok with having a way of turning all the status stuff off.

Reply:
e.g. https://github.com/knative/serving/pull/6813

Reply:
> we might inadvertently convey readiness before we mean to by auto-bumping ObsGen before the conditions are (successfully) updated. 

I am not sure I follow this. Both this and the status of conditions are updated in `updateStatus`. No? This does however mean that reconcilers have to follow the contract of always setting Conditions before returning.

>  I'd be ok with having a way of turning all the status stuff off.

Yes i agree with this.